pages/api/emotionEntries/index.js

```diff
@@ -1,25 +1,42 @@
 import dbConnect from "@/db/connect";
 import EmotionEntries from "@/db/models/emotionEntries";
+import { getServerSession } from "next-auth/next";
+import { authOptions } from "../auth/[...nextauth]";
 
 export default async function handler(request, response) {
+  const session = await getServerSession(request, response, authOptions);
   await dbConnect();
 
   if (request.method === `GET`) {
-    try {
-      const emotionEntries = await EmotionEntries.find();
-      return response.status(200).json(emotionEntries);
-    } catch (error) {
-      console.error(error);
-      response.status(500).json({ error: error.message });
+    if (session) {
```

Comment:
You could simplify the session check by performing it at the beginning of the handler: 

```
  if (!session) {
    response.status(401).json({ status: "Not authorized" });
    return;
  }
```

and then adjusting the code accordingly to avoid repetition.